lib/spack/spack/test/entry_points.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+import sys
+
+import pytest
+
+import spack.config
+import spack.extensions
+
+
+class MockConfigEntryPoint:
+    def __init__(self, tmpdir):
+        self.dir = tmpdir
+        self.name = "mypackage_config"
+
+    def load(self):
+        self.dir.ensure("spack/etc", dir=True)
+        f = os.path.join(self.dir, "spack", "etc", "config.yaml")
+        with open(f, "w") as fh:
+            fh.write("config:\n  install_tree:\n    root: /spam/opt\n")
+
+        def ep():
+            return os.path.join(self.dir, "spack", "etc")
+
+        return ep
+
+
+class MockExtensionsEntryPoint:
+    def __init__(self, tmpdir):
+        self.dir = tmpdir
+        self.name = "mypackage_extensions"
+
+    def load(self):
+        self.dir.ensure("spack/spack-myext/myext/cmd", dir=True)
+        f = os.path.join(self.dir, "spack", "spack-myext", "myext", "cmd", "spam.py")
+        with open(f, "w") as fh:
+            fh.write("description = 'hello world extension command'\n")
+            fh.write("section = 'test command'\n")
+            fh.write("level = 'long'\n")
+            fh.write("def setup_parser(subparser):\n    pass\n")
+            fh.write("def spam(parser, args):\n    print('spam for all!')\n")
+
+        def ep():
+            return os.path.join(self.dir, "spack", "spack-myext")
+
+        return ep
+
+
+def entry_points_factory(tmpdir):
+    def entry_points(group=None):
+        if group == "spack.config":
+            return (MockConfigEntryPoint(tmpdir),)
+        elif group == "spack.extensions":
+            return (MockExtensionsEntryPoint(tmpdir),)
+        return ()
+
+    return entry_points
+
+
+def patch_entry_points(tmpdir, monkeypatch):
```

Comment:
(minor) You can turn this into a fixture, maybe use `tmp_path` instead of `tmpdir` (deprecated):
```suggestion
@pytest.fixture()
def mock_entry_points(tmp_path, monkeypatch):
```